src/Agent/NewRelic/Agent/Extensions/NewRelic.Agent.Extensions/Api/ISegment.cs

```diff
@@ -68,5 +68,12 @@ public interface ISegment : ISpan
         /// Removes this segment from the top of the agent's internal call stack. Should only be used for asynchronous methods. Calling EndSegment is sufficient for synchronous methods.
         /// </summary>
         void RemoveSegmentFromCallStack();
+
+        string GetCategory();
+
+        /// <summary>
+        /// Will be true if a relative end time has been set on the segment.  In most situations, this is only set when a segment is ended.
+        /// </summary>
+        bool IsDone { get; }
```

Comment:
Should we add these to ISegmentExperimental instead?

Reply:
It works both ways, but experimental does give us options so I am going to see about moving it over.